docs/upgrading.md

```diff
@@ -0,0 +1,73 @@
+# Upgrading the Agent
+
+This document provides user guidance around agent upgrades and versioning.
+Like every other piece of software included in your project, the agent 
+should be upgraded frequently enough to pick up ongoing enhancements 
+while allowing room for testing. 
+
+## Watching For New Releases
+
+How can you find out about new releases? The best way is to "Watch" this repo
+from the "Watch" button in the upper-right corner of the main repository 
+page. If you don't want to be notified for every pull request and issue, you can 
+choose "Custom" under "Watch" and select just the "Releases" checkbox.
+
+Another approach is to just check in regularly with the GitHub 
+[releases page](https://github.com/signalfx/splunk-otel-java/releases) for this 
+project. 
+
+## When Does Splunk Release?
+
+Because we are a downstream distribution of [OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation),
+we generally release within a few working days of every upstream release. We also release when we 
+have new features or enhancements that we want to make available.
+
+We contribute to upstream OpenTelemetry and are typically aware of the release cycle
+over there. As a result, depending on the timing of certain patch releases, we may sometimes
+skip a release to reduce the "release thrash" from our distribution.
+
+## When To Upgrade
+
+In general, we recommend that you upgrade to each new version shortly after it is released. Ideally,
+you should upgrade within a few weeks of a release. Given that OpenTelemetry is a very active 
+community, consistent, frequent upgrades will help limit the number of changes between upgrades. 
+
+Upgrades should be intentional and version numbers should be pinned in your build pipeline.
+
+## Upgrade Verification
+
+The Java Instrumentation is a piece of code that runs alongside your code. Because of the rich 
```

Comment:
Just to check: Is its official name "The Java Instrumentation" or "Java Instrumentation"?

I think you should write "code" instead of "a piece of code".

Reply:
> Just to check: Is its official name "The Java Instrumentation" or "Java Instrumentation"?

The repo is `splunk-otel-java` and that's nicer than the title on the main README which says "Splunk Distribution of OpenTelemetry Java" which I think is a mouthful indeed.

> I think you should write "code" instead of "a piece of code".

Sounds good to me.

Reply:
The official name is `Splunk Distribution of OpenTelemetry Java`, I think we also use `the Splunk javaagent` or just `javaagent` in these docs